CMakeLists.txt

```diff
@@ -636,8 +650,30 @@ function(link_threads LIBRARY)
     target_link_libraries(${LIBRARY} Threads::Threads)
 endfunction()
 
+# Deploys extensions to a local repository (a folder structure that contains the duckdb version + binary arch)
+if ("${LOCAL_EXTENSION_REPO}" STREQUAL "")
+  set(LOCAL_EXTENSION_REPO ${CMAKE_BINARY_DIR}/.duckdb/extensions)
```

Comment:
maybe `${CMAKE_BINARY_DIR}/.duckdb/extensions` is not the most intuitive path? I would prefer something like `${CMAKE_BINARY_DIR}/extension_deploy` or `${CMAKE_BINARY_DIR}/.duckdb/extension_repository`

no super strong opinion here, feel free to ignore

Reply:
Problem here is that:
```
D SET extension_directory = 'path/to/some/folder';
D LOAD x;
Error: IO Error: Extension "path/to/some/folder/.duckdb/extensions/fe379c722c/osx_arm64/x.duckdb_extension" not found.
```
where the `.duckdb/extensions` is added by default.

---

My preferred option here would be changing the semantic of extension_directory, where if not specified it defaults to `fs.GetHomeDirectory() + ".duckdb/extension"`, so basically it becomes the base folder of the extension hierarchy.

This will probably break someone's workflows (eg https://github.com/search?q=duckdb+%22SET+extension_directory%22&type=code&p=1), so might need some care, potentially changing name to `extension_base_directory` so an error will be thrown at least.

On the other side, better break stuff today than in a few months...
On this I am not sure, I would push this PR in and in parallel we can set on this semantic (given it's only using extension_directory default, I would argue it's somehow independent).